src/screencast/wlr_screencast.c

```diff
@@ -545,28 +545,90 @@ static void wlr_remove_output(struct xdpw_wlr_output *out) {
 	free(out);
 }
 
+static void wlr_format_modifier_pair_add(struct xdpw_screencast_context *ctx,
+		uint32_t format, uint64_t modifier) {
+	struct xdpw_format_modifier_pair *fm_pair = calloc(1, sizeof(struct xdpw_format_modifier_pair));
+
+	fm_pair->fourcc = format;
+	fm_pair->modifier = modifier;
+
+	logprint(TRACE, "wlroots: format %u (%lu)", fm_pair->fourcc, fm_pair->modifier);
+
+	wl_list_insert(&ctx->format_modifier_pairs, &fm_pair->link);
+
+}
+
+static void wlr_format_modifier_pair_emtpy_list(struct xdpw_screencast_context *ctx) {
+	struct xdpw_format_modifier_pair *fm_pair, *tmp_fmp;
+	wl_list_for_each_safe(fm_pair, tmp_fmp, &ctx->format_modifier_pairs, link) {
+		wl_list_remove(&fm_pair->link);
+		free(fm_pair);
+	}
+}
+
 static void linux_dmabuf_handle_modifier(void *data,
 		struct zwp_linux_dmabuf_v1 *zwp_linux_dmabuf_v1,
 		uint32_t format, uint32_t modifier_hi, uint32_t modifier_lo) {
 	struct xdpw_screencast_context *ctx = data;
 
 	logprint(TRACE, "wlroots: linux_dmabuf_handle_modifier called");
 
-	struct xdpw_format_modifier_pair *fm_pair = calloc(1, sizeof(struct xdpw_format_modifier_pair));
-
-	fm_pair->fourcc = format;
-	fm_pair->modifier = ((((uint64_t)modifier_hi) << 32) | modifier_lo);
-
-	logprint(TRACE, "wlroots: format %u (%u)", fm_pair->fourcc, fm_pair->modifier);
-
-	wl_list_insert(&ctx->format_modifier_pairs, &fm_pair->link);
+	wlr_format_modifier_pair_add(ctx, format, ((((uint64_t)modifier_hi) << 32) | modifier_lo));
```

Comment:
Nit: there is an unnecessary parenthesis here. Maybe just use an intermediate `uint64_t modifier` variable to avoid the parenthesis soup?

Reply:
done